### DISCUSSION TITLE
When deploying Next.js, what to do with node_modules?

### DISCUSSION THREAD

**Discussion by jordyvandomselaar** [vercel/next.js#15371] [Help]
We're deploying a Next.js app to production with `npm run build && npm run start`, I can imagine we need `node_modules` for the Next server but we don't need them all, packages like `Formik` aren't needed after build right? Do we move these to devdependencies?

**Comment by jamesmosier** [comment]
My personal workflow is to install all `dependencies` in my `package.json` on my production server. The reason because I think of `devDependencies` as truly only needed for dev. It also helps to reduce any errors/issues by having to try to determine which dependencies are needed on the client vs. server. 

So I'd recommend installing your node_modules on your server in your production environments and keep your `devDependencies` as just that, dependencies for development.

**Comment by uguremirmustafa** [comment]
Next team tries to force us developers to use Vercel. Any time I need to handle deployment of a Next.js app on a custom server, I get tons of headaches.
